### ISSUE TITLE
Geometry.toString() throws NegativeArraySizeException

### ISSUE THREAD

**Issue by stormof** [locationtech/jts#515]
WKTWriter.stringOfChar() can throw NegativeArraySizeException after this commit
https://github.com/locationtech/jts/commit/b464d54960cf0c99195a0f93a26ccfeab55c7f55#diff-c75b976078d9531815da8d21799c9bc0.
It happens when precision model  is Fixed with scale < 0.01. 
For example:
scale = 0.009
1 + (int) Math.ceil(Math.log(0.009) / Math.log(10)) = -1
new StringBuilder(-1 ) throws NegativeArraySizeException

In our project, we have a huge number of logs with geometries and we have to remove them due to this exception.

**Comment by dr-jts** [comment]
I can't reproduce this in `master` using the following test code:
```java
  public void testFixedSmallFactor() {
    double scale = 0.009;
    PrecisionModel pm = new PrecisionModel(scale);
    GeometryFactory geometryFactory = new GeometryFactory(pm);
    Coordinate[] coordinates = { new Coordinate(1000, 1000),
        new Coordinate(2000, 2000) };
    LineString line = geometryFactory.createLineString(coordinates);
    WKTWriter writer = new WKTWriter();
    String wkt = writer.write(line);
    System.out.println(wkt);
  }
```
Is the test correct?  
Perhaps this was fixed by a subsequent change?

**Comment by stormof** [comment]
This is releated to Fixed PrecisionModel:
  public int getMaximumSignificantDigits() {
    int maxSigDigits = 16;
    if (modelType == FLOATING) {
      maxSigDigits = 16;
    } else if (modelType == FLOATING_SINGLE) {
      maxSigDigits = 6;
    } else if (modelType == FIXED) {
      maxSigDigits = 1 + (int) Math.ceil(Math.log(getScale()) / Math.log(10));
    }
    return maxSigDigits;
  }

**Comment by stormof** [comment]
JDK 11 the same  error:
![image](https://user-images.githubusercontent.com/42542770/73428995-3805de80-4343-11ea-8d35-5604b70fd6b0.png)

**Comment by dr-jts** [comment]
> It seems this issue was fixed a year ago in [a8b86cb#diff-c75b976078d9531815da8d21799c9bc0]
> 
> So, we need to wait JTS 1.16.2?

I'm afraid so - or more likely 1.17.
